src/main/java/org/testng/TestNG.java

```diff
@@ -1226,60 +1236,63 @@ private void generateReports(List<ISuite> suiteRunners) {
-        Version.displayBanner();
-      }
 
-      // First initialize the suite runners to ensure there are no configuration issues.
-      // Create a map with XmlSuite as key and corresponding SuiteRunner as value
+    if (m_suites.get(0).getVerbose() >= 2) {
+      Version.displayBanner();
+    }
+
+    // First initialize the suite runners to ensure there are no configuration issues.
+    // Create a map with XmlSuite as key and corresponding SuiteRunner as value
+    for (XmlSuite xmlSuite : m_suites) {
+      createSuiteRunners(suiteRunnerMap, xmlSuite);
+    }
+
+    //
+    // Run suites
+    //
+    if (m_suiteThreadPoolSize == 1 && !m_randomizeSuites) {
+      // Single threaded and not randomized: run the suites in order
       for (XmlSuite xmlSuite : m_suites) {
-        createSuiteRunners(suiteRunnerMap, xmlSuite);
+        runSuitesSequentially(xmlSuite, suiteRunnerMap, getVerbose(xmlSuite),
+                getDefaultSuiteName());
       }
-
       //
-      // Run suites
+      // Generate the suites report
       //
-      if (m_suiteThreadPoolSize == 1 && !m_randomizeSuites) {
-        // Single threaded and not randomized: run the suites in order
-        for (XmlSuite xmlSuite : m_suites) {
-          runSuitesSequentially(xmlSuite, suiteRunnerMap, getVerbose(xmlSuite),
-              getDefaultSuiteName());
-        }
-      } else {
-        // Multithreaded: generate a dynamic graph that stores the suite hierarchy. This is then
-        // used to run related suites in specific order. Parent suites are run only
-        // once all the child suites have completed execution
-        DynamicGraph<ISuite> suiteGraph = new DynamicGraph<>();
-        for (XmlSuite xmlSuite : m_suites) {
-          populateSuiteGraph(suiteGraph, suiteRunnerMap, xmlSuite);
-        }
-
-        IThreadWorkerFactory<ISuite> factory = new SuiteWorkerFactory(suiteRunnerMap,
-          0 /* verbose hasn't been set yet */, getDefaultSuiteName());
-        GraphThreadPoolExecutor<ISuite> pooledExecutor =
-                new GraphThreadPoolExecutor<>("suites", suiteGraph, factory, m_suiteThreadPoolSize,
-                        m_suiteThreadPoolSize, Integer.MAX_VALUE, TimeUnit.MILLISECONDS,
-                        new LinkedBlockingQueue<Runnable>());
-
-        Utils.log("TestNG", 2, "Starting executor for all suites");
-        // Run all suites in parallel
-        pooledExecutor.run();
-        try {
-          pooledExecutor.awaitTermination(Long.MAX_VALUE, TimeUnit.MILLISECONDS);
-          pooledExecutor.shutdownNow();
-        }
-        catch (InterruptedException handled) {
-          Thread.currentThread().interrupt();
-          error("Error waiting for concurrent executors to finish " + handled.getMessage());
-        }
-      }
-    }
-    else {
-      setStatus(HAS_NO_TEST);
-      error("No test suite found. Nothing to run");
-      usage();
+      return Lists.newArrayList(suiteRunnerMap.values());
```

Comment:
`Collections.unmodifiableList(...)`?

Reply:
`suiteRunnerMap.values()` returns a `Collection<ISuite>` and `Collections.unmodifiableList()` takes in a `List` and not a `Collection`. So the only choice I have would be to pass the `Lists.newArrayList()` call to `Collections.unmodifiableList()`. Is that required ?

Reply:
Ok, forget it :)